client/dashboard/components/site-preview-link/index.tsx

```diff
@@ -10,38 +9,77 @@ const Separator = styled.span( {
 	},
 } );
 
+const ONE_HOUR_IN_MILLISECONDS = 60 * 60 * 1000;
+const ONE_DAY_IN_MILLISECONDS = 24 * ONE_HOUR_IN_MILLISECONDS;
+const ONE_MINUTE_IN_SECONDS = 60;
+const ONE_HOUR_IN_SECONDS = 60 * ONE_MINUTE_IN_SECONDS;
+const ONE_DAY_IN_SECONDS = 24 * ONE_HOUR_IN_SECONDS;
+
 type LinkExpiryCopyProps = {
 	expiresAt: string;
 };
 
-const LinkExpiryCopy = ( { expiresAt }: LinkExpiryCopyProps ) => {
-	const now = moment();
-	const expiryDate = moment( expiresAt );
+function getExpiryStringLessThanOneDay( expiryDate: Date, now: Date ) {
+	const expiryInSeconds = Math.round( ( expiryDate.getTime() - now.getTime() ) / 1000 );
 
-	const difference = expiryDate.diff( now );
+	if ( expiryInSeconds < ONE_MINUTE_IN_SECONDS ) {
+		return sprintf(
+			// translators: %(seconds) is the number of seconds until the link expires.
+			_n( 'Expires in %(seconds)d second', 'Expires in %(seconds)d seconds', expiryInSeconds ),
+			{ seconds: expiryInSeconds }
+		);
+	}
 
-	if ( difference < 0 ) {
-		return __( 'Expired.' );
+	if ( expiryInSeconds < ONE_HOUR_IN_SECONDS ) {
+		const minutes = Math.round( expiryInSeconds / ONE_MINUTE_IN_SECONDS );
```

Comment:
Should all `Math.round()` in this PR be `Math.floor()` instead?

For example, in this case, if `expiryInSeconds` is 59 minutes and 40 seconds, the string will be `Expires in 60 minutes`. (Do we want this?)

Reply:
Oh OK, in this case I think it's OK :)